### ISSUE TITLE
Do not return TRUE from backdrop_session_regenerate()

### ISSUE THREAD

**Issue by kiamlaluno** [backdrop/backdrop-issues#6019]
[`backdrop_session_regenerate()`](https://docs.backdropcms.org/api/backdrop/core%21includes%21session.inc/function/backdrop_session_regenerate/1) returns `TRUE`, but none of the functions calling it uses the returned value, which is not even documented.
Even [`drupal_session_regenerate()`](https://api.drupal.org/api/drupal/includes%21session.inc/function/drupal_session_regenerate/7.x), the function that in Backdrop is just a wrapper for `backdrop_session_regenerate()` returns `TRUE` or `FALSE`.

The first `return TRUE;` in `backdrop_session_regenerate()` should be changed to `return;`, while the other ones should be removed.

**Comment by kiamlaluno** [comment]
Tests randomly fail, but those failures are not related to this change.

**Comment by klonos** [comment]
> CSpell reports words to change, but those words are not in the file I changed.

Have you tried rebasing your PR branch with current 1.x?

> Furthermore, some of those words are already correct, like HMAC; it is the dictionary that should be updated.

That's right. Each PR that introduces new words needs to add these words (in alphabetical order) in the https://github.com/backdrop/backdrop/blob/1.x/.cspell/backdrop.dic file. We can create a new issue about these words, file a PR, and once it is merged you'll be able to rebase your PR branch and it shouldn't complain any more. But I'm thinking why go through all the trouble when we can fix things in this PR? Your call @kiamlaluno: either fix these in your PR, or raise a separate issue about them.

**Comment by kiamlaluno** [comment]
@klonos The [issue_6019](https://github.com/kiamlaluno/backdrop/tree/issue_6019) branch is already synchronized with up-stream 1.x branch.

**Comment by kiamlaluno** [comment]
Also, notice that *HMAC* has not been added in my PR; it is a word that already exists in the repository, not even in a file my PR changed. I apologize if I did not made this clear.

**Comment by klonos** [comment]
@kiamlaluno I've filed a PR to handle the scpell notices in #6047 😉

**Comment by quicksketch** [comment]
Thanks @kiamlaluno for the explanation and PR! Everything lines up as you say. I merged https://github.com/backdrop/backdrop/pull/4376 into 1.x and 1.25.x. Thanks!
